src/sagemaker/workflow/steps.py

```diff
@@ -620,7 +632,17 @@ def arguments(self) -> RequestType:
                     vpc_config=self.model.vpc_config,
                     enable_network_isolation=self.model.enable_network_isolation(),
                 )
-        request_dict.pop("ModelName", None)
+
+        # Continue to pop custom prefix if not explicitly opted-in
+        if not _pipeline_config or not _pipeline_config.use_custom_job_prefix:
+            request_dict.pop("ModelName", None)
+        else:
+            # Strip timestamp -- workflow will use only the base-job name for name generation
```

Comment:
Let's move this into a separate function and put in utils.py. We will re-use for all the steps.

Reply:
Yep was actually in the process of doing that before I saw this comment. Sounds good.